projects/schematics/src/shared/utils/feature-utils.ts

```diff
@@ -484,3 +484,29 @@ function createDependentFeaturesLog(
     ', '
   )}: ${notSelectedFeatures.join(', ')}\n`;
 }
+
+/**
+ * This function is a workaround for the GitHub issue
+ * https://github.com/angular/angular-cli/issues/16320.
+ * Using `ng add` with options of type array (--features)
+ * may not work correctly in some situations, and might only work on the
+ * second attempt.
+ */
+export function normalizeOptionsFeatures<OPTIONS extends LibraryOptions>(
+  options: OPTIONS
+): string[] {
+  return convertToArray(options.features ?? []);
+}
+
+export function convertToArray(values: string[] | string): string[] {
+  let optionsArray: string[] = [];
+  if (Array.isArray(values)) {
+    optionsArray = values;
+  } else if (values.includes(',')) {
+    optionsArray = values.split(',');
```

Comment:
Question: shall we somehow trim whitespaces, if using as `--features="Cart, Order, Checkout"` (with spaces)?
1. we should test it manually, if it works now (wihtout trimming)
2. if it doesn't work with spaces, then let's fix it + add unit tests for this case

Reply:
My initial thought was that the correct command is --features=Cart,Order,Checkout, as this is the syntax used for an array-type option. The command you provided assumes that the feature is a string. However, the type array | string is just a workaround. I believe we should still treat it as an array.

I haven't tested the whitespace theory yet.